plugins/services/src/js/components/modals/CreateServiceModalForm.tsx

```diff
@@ -307,46 +324,30 @@ class CreateServiceModalForm extends React.Component {
 
     this.setState(newState);
   };
-  handleFormChange = (event) => {
-    const fieldName = event.target.getAttribute("name");
-    if (!fieldName) {
-      return;
-    }
 
-    let { batch } = this.state;
-    let value = event.target.value;
-    if (event.target.type === "checkbox") {
-      value = event.target.checked;
-    }
-    const path = fieldName.split(".");
-    batch = batch.add(new Transaction(path, value));
+  addTransaction = (t) => {
+    const batch = this.state.batch.add(t);
+    this.setState({ appConfig: this.getAppConfig(batch), batch });
+  };
 
-    const newState = {
-      appConfig: this.getAppConfig(batch),
-      batch,
-    };
+  handleFormChange = ({ target }) => {
+    if (!target.name) {
+      return;
+    }
 
-    this.setState(newState);
+    const path = target.name.split(".");
+    const value = target.type === "checkbox" ? target.checked : target.value;
```

Comment:
`event.target.value` works the same as `event.target.getAttribute("name")` in all browser. additionally it widens our interface for manually crafted events. formerly we got native browser-events for each change that has happened on an input. in the CSI-section we now send events explicitly with exactly the content we want.